### PR TITLE
Dlm/allowed satellite

### PR DESCRIPTION
**Pull request by dlamarmorgan**

### CODE REVIEW

File: pkg/recipe/entrypoint.sh [hunk 1]
```diff
@@ -32,6 +32,11 @@ if [ "$STORJ_ROLE" == "satellite-api" ]; then
   fi
 fi
 
+if [ "$STORJ_ROLE" == "authservice" ]; then
```
Comment (dlamarmorgan) [inline pkg/recipe/entrypoint.sh:35]: one potential drawback of this approach, is changing the satellite now requires updating the entrypoint,sh file inside the container, which is not at easy as updating an environment variable.

Maybe this code block can be changed to use the old variable in some way and just resolve the hostname portion.

File: cmd/files/docker/build.Dockerfile [hunk 2]
```diff
@@ -26,7 +26,7 @@ RUN git fetch https://review.dev.storj.io/storj/storj ${REF} && git checkout FET
 
 FROM ${TYPE} AS binaries
 RUN env env GO111MODULE=on GOOS=js GOARCH=wasm GOARM=6 -CGO_ENABLED=1 TAG=head scripts/build-wasm.sh && \
-    go build ./cmd/... && \
+    go build -buildvcs=false ./cmd/... && \
```
Comment (elek) [inline cmd/files/docker/build.Dockerfile:29]: Wondering what is the reason for this change. I know that we have an own solution but it can be useful anyway... Couldn't be?

File: cmd/files/docker/edge.Dockerfile [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ARG TYPE
 
-FROM ghcr.io/elek/storj-build:20220216-1 AS base
+FROM ghcr.io/dlamarmorgan/storj-build:20220725-1 AS base
```
Comment (elek) [inline cmd/files/docker/edge.Dockerfile:3]: Let's use a common ghcr.io/storj or img.dev.storj.io...

File: pkg/recipe/entrypoint.sh (hunk 1 above)
Comment (elek) [reply pkg/recipe/entrypoint.sh:35]: What is the motivation to move the `STORJ_ALLOWED_SATELLITES` here?

Using the auth service from the host machine, maybe?

Comment (elek) [reply pkg/recipe/entrypoint.sh:35]: Just checked and this is what we have in `peer.go`:

```
if len(config.AllowedSatellites) == 0 {
   return nil, errs.New("allowed satellites parameter '--allowed-satellites' is required")
}
```

The default is always our URL, so it might be safe to accept a magic `ALL` or `0.0.0.0` string to not restrict satellites at all.

**Comment by dlamarmorgan** [comment]
retesting changes and moving review to Gerrit.
